### ISSUE TITLE
DEPR: drop support for "10h36m28.5s" formats

### ISSUE THREAD

**Issue by jbrockmendel** [dateutil/dateutil#886]
As far as I can tell these are a legacy format really intended to represent timedeltas, really don't belong in a datetime parser.  They add non-trivial complexity to the parser code.

**Comment by pganssle** [comment]
Yeah I'm really not a big fan of these. I think we'll need to write a new `timedelta` parser for these maybe, which is blocked on a whole bunch of other stuff.
